bika/lims/browser/bika_listing.py

```diff
@@ -1367,18 +1347,101 @@ def rendered_items(self):
         self.cat = self.request.get('ajax_category_expand')
         self.contentFilter[self.category_index] = self.request.get('cat')
 
-        # selected review_state must be adhered to
-        st_id = self.request.get('review_state')
-
         # These are required to allow the template to work with this class as
         # the view.  Normally these are attributes of class BikaListingTable.
         self.bika_listing = self
         self.this_cat_selected = True
         self.this_cat_batch = self.folderitems()
 
-        data = self.render_items()
+        data = self.render_items(self.context)
         return data
 
+    def get_transitions_for_items(self, items):
+        """Extract Worfklow transitions for the bika listing items
+        """
+        workflow = get_tool('portal_workflow')
+        out = {}
+
+        # helper method to extract the object from an bika listing item
+        def get_object_from_item(item):
+            brain_or_object = item.get("obj")
+            return api.get_object(brain_or_object)
+
+        # extract all objects from the items
+        objects = map(get_object_from_item, items)
```

Comment:
I'll do this:

        # extract all objects from the items
        objects = map(get_object, [item.get('obj') for item in items])

Because if any item actually doesn't have an object, there will be a traceback from many lines earlier.